recipes/ffmpeg/all/conanfile.py

```diff
@@ -0,0 +1,352 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import glob
+import shutil
+
+
+class FFMpegConan(ConanFile):
+    name = "ffmpeg"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A complete, cross-platform solution to record, convert and stream audio and video"
+    # https://github.com/FFmpeg/FFmpeg/blob/master/LICENSE.md
+    license = ("LGPL-2.1-or-later", "GPL-2.0-or-later")
+    homepage = "https://ffmpeg.org"
+    topics = ("ffmpeg", "multimedia", "audio", "video", "encoder", "decoder", "encoding", "decoding",
+             "transcoding", "multiplexer", "demultiplexer", "streaming")
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False],
+               "fPIC": [True, False],
+               "postproc": [True, False],
+               "zlib": [True, False],
+               "bzlib": [True, False],
+               "lzma": [True, False],
+               "iconv": [True, False],
+               "freetype": [True, False],
+               "openjpeg": [True, False],
+               "openh264": [True, False],
+               "opus": [True, False],
+               "vorbis": [True, False],
+               "zmq": [True, False],
+               "sdl2": [True, False],
+               "x264": [True, False],
+               "x265": [True, False],
+               "vpx": [True, False],
+               "mp3lame": [True, False],
+               "fdk_aac": [True, False],
+               "webp": [True, False],
+               "ssl": [False, "openssl", "securetransport"],
+               "alsa": [True, False],
+               "pulse": [True, False],
+               "vaapi": [True, False],
+               "vdpau": [True, False],
+               "xcb": [True, False],
+               "appkit": [True, False],
+               "avfoundation": [True, False],
+               "coreimage": [True, False],
+               "audiotoolbox": [True, False],
+               "videotoolbox": [True, False],
+               "qsv": [True, False]}
```

Comment:
I think the names of all options should be reviewed.
Some/all of them should be prepended with `with_`.
so e.g. `zmq` should become `with_zmq`.